### PR TITLE
Add e2e tests

### PR DESCRIPTION
**Pull request by sgalsaleh**
#### What this PR does / why we need it:

Adds E2E tests using the C11y matrix for a variety of Kubernetes distributions.

#### Which issue(s) this PR fixes:

Fixes [SC-86765](https://app.shortcut.com/replicated/story/86765/testing-strategy-for-replicated-sdk)

#### Special notes for your reviewer:

Added a `test-chart`  that is dynamically packaged in the CI workflow to include the built replicated-sdk as a subchart. It is then used to create a replicated release and install it via the replicated registry.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Does this PR require documentation?

NONE

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -28,24 +35,181 @@ jobs:
         run: |
           make can-i-deploy || echo "::warning:: can-i-deploy says no; provider(s) must successfully verify before release"
 
-  make-build:
-    runs-on: ubuntu-22.04
+  build-and-push-e2e:
+    runs-on: ubuntu-20.04
```
[the remark this answers is not in the document]
Comment (sgalsaleh) [reply .github/workflows/main.yaml:39]: no reason. left by mistake while testing something... good catch

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -28,24 +35,181 @@ jobs:
...
+      - name: Create cluster
+        id: create-cluster 
+        uses: replicatedhq/replicated-actions/create-cluster@v1.0.1
+        with:
+          api-token: ${{ secrets.C11Y_MATRIX_TOKEN }}
+          kubernetes-distribution: ${{ matrix.cluster.distribution }}
+          kubernetes-version: ${{ matrix.cluster.version }}
+          cluster-name: automated-${{ github.run_id }}-${{ matrix.cluster.distribution }}-${{ matrix.cluster.version }}
+          ttl: 2h
+          export-kubeconfig: true
```
[the remark this answers is not in the document]
Comment (sgalsaleh) [reply .github/workflows/main.yaml:149]: problem is one of the variations is openshift, which can take up-to 15 minutes? that leaves 5 minutes for the queue period, which i think is reasonable in certain scenarios.

File: test.sh
Comment (cbodonnell) [inline test.sh:1]: is this used in the e2e testing?

File: .github/workflows/main.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (cbodonnell) [reply .github/workflows/main.yaml:149]: would a developer have to re-run the test if it's queued for more than 5 minutes?

[the remark this answers is not in the document]
Comment (sgalsaleh) [reply .github/workflows/main.yaml:149]: for now, yes.

**Review by cbodonnell — APPROVED** [review]

**Review by cbodonnell — DISMISSED** [review]

**Review by cbodonnell — APPROVED** [review]

**Pull request merged**